plugins/woocommerce-blocks/packages/components/label/index.tsx

```diff
@@ -45,7 +47,17 @@ const Label = ( {
 	if ( hasLabel && hasScreenReaderLabel && label !== screenReaderLabel ) {
 		return (
 			<Wrapper { ...wrapperProps }>
-				<span aria-hidden="true">{ label }</span>
+				{ allowHTML ? (
+					<span
+						aria-hidden="true"
+						dangerouslySetInnerHTML={ {
+							__html: label,
```

Comment:
How about doing some form of sanitization using `sanitizeHTML`? This uses DomPurify library.